cylc/flow/task_events_mgr.py

```diff
@@ -843,10 +857,10 @@ def _custom_handler_callback(self, ctx, schd, id_key):
         else:
             self.unset_waiting_event_timer(id_key)
 
-    def _db_events_insert(self, itask, event="", message=""):
+    def _db_events_insert(self, itask, event_time, event="", message=""):
         """Record an event to the DB."""
         self.workflow_db_mgr.put_insert_task_events(itask, {
-            "time": get_current_time_string(),
+            "time": event_time,
```

Comment:
I can't see any reason that this should be got at the time of execution and not fed down from the calling method. Without this change, however, the event time in the table may not match the event time returned by poll.

Reply:
IMO this is a bugfix!

Reply:
What is `event_time`, is it an ISO8601 string, or an integer epoch time? I've gotta feeling that it might be both (search for now in the diff).

Reply:
It _is_ _either_ - 

> IMO this is a bugfix!

I was hoping that was how you'd see it.